test/integration/gardenlet/backupbucket/backupbucket_test.go

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupbucket_test
+
+import (
+	"fmt"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("BackupBucket controller tests", func() {
+	var (
+		backupBucket                      *gardencorev1beta1.BackupBucket
+		generatedSecret                   *corev1.Secret
+		coreGeneratedSecret               *corev1.Secret
+		extensionSecret                   *corev1.Secret
+		generatedSecretRef                *corev1.SecretReference
+		coreGeneratedSecretRef            *corev1.SecretReference
+		extensionBackupBucket             *extensionsv1alpha1.BackupBucket
+		expectedExtensionBackupBucketSpec extensionsv1alpha1.BackupBucketSpec
+
+		backupBucketReady = func(makeReady bool) {
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(extensionSecret), extensionSecret)).To(Succeed())
+				g.Expect(extensionSecret.Data).To(Equal(secret.Data))
+
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(extensionBackupBucket), extensionBackupBucket)).To(Succeed())
+				g.Expect(extensionBackupBucket.Spec).To(Equal(expectedExtensionBackupBucketSpec))
```

Comment:
Why do we need this line? We only care about the operation annotation and the status, or?

Reply:
We set the spec here, https://github.com/gardener/gardener/blob/df2e70021bd66c5713ec25478a24b29dfe3a3336/pkg/gardenlet/controller/backupbucket/actuator.go#L199-L222
Thought there is no harm in checking.